src/main/java/ScheduleLinkedList.java

```diff
@@ -1,47 +1,106 @@
-import java.util.ArrayList;
 import java.util.List;
+import java.util.ArrayList;
 
-public class ScheduleLinkedList {
 
+public class ScheduleLinkedList {
     private ScheduleNode head;
 
-    public ScheduleLinkedList(){
-
+    public ScheduleLinkedList() {
+        this.head = null;
     }
 
-    public void addFirst(ScheduleNode scheduleNode){
-
+    public void addFirst(ScheduleNode scheduleNode) {
+        if (scheduleNode != null) {
+            scheduleNode.setNext(head);
```

Comment:
the new scheduleNode should still point to null, otherwise we'll have a loop starting at the head